src/util/TileCellData.js

```diff
@@ -182,11 +189,29 @@ export default class TileCellData {
       if (cell_area > 10) {
         // fit an ellipse
         const rotatedRect = cv.fitEllipse(contours.get(i));
-        // add ellipse to data
-        this.add_cell(rotatedRect.center.x, rotatedRect.center.y, rotatedRect.size.width, rotatedRect.size.height, rotatedRect.angle);
+
+        // Need to map point (in tile coords) to annotation (in % of image)
+        let scale = this.tile.bounds.height / this.tile.sourceBounds.height;
+
+        // let shift_x = (this.tile.x) * this.tile.bounds.width;
+        // let shift_y = (this.tile.y) * this.tile.bounds.height;
+        let shift_x = this.tile.x * this.tile.bounds.x / this.tile.x;
+        let shift_y = this.tile.y * this.tile.bounds.y / this.tile.y;
+
+        // test if ellipse centroid is contained in the annotation polygon
```

Comment:
this comment doesn't refer to the line below, probably should be just above the `if` statement I think